From c3c5e3a970fa94486c0af51812f84bec52fa12ad Mon Sep 17 00:00:00 2001
From: Juan Quintela <quintela@redhat.com>
Date: Sun, 9 Oct 2011 19:32:08 +0200
Subject: [PATCH 4/8] savevm: define qemu_get_byte() using qemu_peek_byte()

RH-Author: Juan Quintela <quintela@redhat.com>
Message-id: <f6780a4802efc0842cdab1d70b59750f5c266429.1318188414.git.quintela@redhat.com>
Patchwork-id: 33953
O-Subject: [PATCH 3/5] savevm: define qemu_get_byte() using qemu_peek_byte()
Bugzilla: 725565
RH-Acked-by: Orit Wasserman <owasserm@redhat.com>
RH-Acked-by: Paolo Bonzini <pbonzini@redhat.com>
RH-Acked-by: Alex Williamson <alex.williamson@redhat.com>

Signed-off-by: Juan Quintela<quintela@redhat.com>
Signed-off-by: Juan Quintela <quintela@redhat.com>
---
 savevm.c |   15 ++++++---------
 1 files changed, 6 insertions(+), 9 deletions(-)

Signed-off-by: Michal Novotny <minovotn@redhat.com>
---
 savevm.c |   15 ++++++---------
 1 files changed, 6 insertions(+), 9 deletions(-)

diff --git a/savevm.c b/savevm.c
index a7e2c0d..f25c5c7 100644
--- a/savevm.c
+++ b/savevm.c
@@ -581,17 +581,14 @@ static int qemu_peek_byte(QEMUFile *f)
 
 int qemu_get_byte(QEMUFile *f)
 {
-    if (f->is_write) {
-        abort();
-    }
+    int result;
 
-    if (f->buf_index >= f->buf_size) {
-        qemu_fill_buffer(f);
-        if (f->buf_index >= f->buf_size) {
-            return 0;
-        }
+    result = qemu_peek_byte(f);
+
+    if (f->buf_index < f->buf_size) {
+        f->buf_index++;
     }
-    return f->buf[f->buf_index++];
+    return result;
 }
 
 int64_t qemu_ftell(QEMUFile *f)
-- 
1.7.4.4

